tests/resource/test_resource.py

```diff
@@ -153,13 +190,39 @@ def _construct_tape_from_ops(lst_ops):
         ],
         100,
     ),
+    ([qml.Hadamard(0), qml.CNOT([0, 1]), _CustomOpWithResource(wires=[1, 0])], 0),
+    (
+        [
+            qml.PauliZ(0),
+            qml.CNOT([0, 1]),
+            qml.RX(1.23, 2),
+            _CustomOpWithResource(wires=[0, 2]),
+            _CustomOpWithoutResource(wires=[0, 1]),
+        ],
+        10,
+    ),
+    (
+        [
+            qml.Hadamard(0),
+            qml.RX(1.23, 1),
+            qml.CNOT([0, 1]),
+            qml.RX(4.56, 1),
+            qml.Hadamard(0),
+            qml.Hadamard(1),
+            _CustomOpWithoutResource(wires=[0, 1]),
+        ],
+        100,
+    ),
 )
 
 resources_data = (
     Resources(),
     Resources(2, 2, {"Hadamard": 1, "CNOT": 1}, 2, 0),
     Resources(3, 3, {"PauliZ": 1, "CNOT": 1, "RX": 1}, 2, 10),
     Resources(2, 6, {"Hadamard": 3, "RX": 2, "CNOT": 1}, 4, 100),
+    Resources(2, 5, {"Hadamard": 1, "CNOT": 1, "Identity": 1, "PauliZ": 2}, 5, 0),
```

Comment:
Good to mention what these items are. i.e.,
`#  wires, gates, gate_types, depth, shots`